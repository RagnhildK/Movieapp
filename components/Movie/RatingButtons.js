import React from "react";
import { View, Button } from "react-native";
import { useDispatch, useSelector } from "react-redux";
import { rateMovie } from "../../redux/movieSlicer";
import { Rating } from "react-native-ratings";

function RatingButtons({ id }) {
  const { movies } = useSelector((state) => state.movieRatings);
  const dispatch = useDispatch();

  const handlePress = (rating) => {
    dispatch(rateMovie({ rating: rating, id: id }));
  };

  return (
    <View className="MovieItem">
      <Rating></Rating>
    </View>
  );
}

export default RatingButtons;
