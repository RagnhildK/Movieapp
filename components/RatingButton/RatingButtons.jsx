import React from "react";
import { StyleSheet, View } from "react-native";
import { useDispatch, useSelector } from "react-redux";
import { rateMovie } from "../../redux/movieSlicer";
import { AirbnbRating } from "react-native-ratings";
import { Colors } from "../../styles/colors";

function RatingButtons({ id }) {
  const { ratings } = useSelector((state) => state.movieRatings);
  const dispatch = useDispatch();

  const ratingCompleted = (rating) => {
    dispatch(rateMovie({ rating: rating, id: id }));
  };

  return (
    <View style={styles.movieRating}>
      <AirbnbRating
        onFinishRating={(rating) => ratingCompleted(rating)}
        defaultRating={ratings[id]}
        showRating={false}
        size={25}
      />
    </View>
  );
}
const styles = StyleSheet.create({
  movieRating: {
    flex: 1,
    justifyContent: "center",
  },
});

export default RatingButtons;
