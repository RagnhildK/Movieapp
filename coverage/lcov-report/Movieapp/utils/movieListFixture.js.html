
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for Movieapp/utils/movieListFixture.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> / <a href="index.html">Movieapp/utils</a> movieListFixture.js</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>2/2</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>0/0</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>0/0</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>2/2</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a>
<a name='L28'></a><a href='#L28'>28</a>
<a name='L29'></a><a href='#L29'>29</a>
<a name='L30'></a><a href='#L30'>30</a>
<a name='L31'></a><a href='#L31'>31</a>
<a name='L32'></a><a href='#L32'>32</a>
<a name='L33'></a><a href='#L33'>33</a>
<a name='L34'></a><a href='#L34'>34</a>
<a name='L35'></a><a href='#L35'>35</a>
<a name='L36'></a><a href='#L36'>36</a>
<a name='L37'></a><a href='#L37'>37</a>
<a name='L38'></a><a href='#L38'>38</a>
<a name='L39'></a><a href='#L39'>39</a>
<a name='L40'></a><a href='#L40'>40</a>
<a name='L41'></a><a href='#L41'>41</a>
<a name='L42'></a><a href='#L42'>42</a>
<a name='L43'></a><a href='#L43'>43</a>
<a name='L44'></a><a href='#L44'>44</a>
<a name='L45'></a><a href='#L45'>45</a>
<a name='L46'></a><a href='#L46'>46</a>
<a name='L47'></a><a href='#L47'>47</a>
<a name='L48'></a><a href='#L48'>48</a></td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { configureStore } from "@reduxjs/toolkit";
import movieRatingReducer from "../redux/movieSlicer"
&nbsp;
const movies = {
    425909: {
      title: "Ghostbusters: Afterlife",
      posterPath: "/sg4xJaufDiQl7caFEskBtQXfD4x.jpg",
      overview:
        "When a single mom and her two kids arrive in a small town, they begin to discover their connection to the original Ghostbusters and the secret legacy their grandfather left behind.",
    },
    438695: {
      title: "Sing 2",
      posterPath: "/aWeKITRFbbwY8txG5uCj4rMCfSP.jpg",
      overview:
        "Buster and his new cast now have their sights set on debuting a new show at the Crystal Tower Theater in glamorous Redshore City. But with no connections, he and his singers must sneak into the Crystal Entertainment offices, run by the ruthless wolf mogul Jimmy Crystal, where the gang pitches the ridiculous idea of casting the lion rock legend Clay Calloway in their show. Buster must embark on a quest to find the now-isolated Clay and persuade him to return to the stage.",
    },
    460458: {
      title: "Resident Evil: Welcome to Raccoon City",
      posterPath: "/7uRbWOXxpWDMtnsd2PF3clu65jc.jpg",
      overview:
        "Once the booming home of pharmaceutical giant Umbrella Corporation, Raccoon City is now a dying Midwestern town. The company’s exodus left the city a wasteland…with great evil brewing below the surface. When that evil is unleashed, the townspeople are forever…changed…and a small group of survivors must work together to uncover the truth behind Umbrella and make it through the night.",
    },
    512195: {
      title: "Red Notice",
      posterPath: "/lAXONuqg41NwUMuzMiFvicDET9Y.jpg",
      overview:
        "An Interpol-issued Red Notice is a global alert to hunt and capture the world's most wanted. But when a daring heist brings together the FBI's top profiler and two rival criminals, there's no telling what will happen.",
    },
  };
&nbsp;
  export const mockStore = configureStore({
    reducer: {
      movieRatings: movieRatingReducer,
    },
    preloadedState: {
      movieRatings: {
        sessionID: "",
        username: "",
        nmbMovies: 0,
        loading: false,
        movies: movies,
        ratings: {},
        totalResults: {},
        sortedIDs: [],
        participants: [],
      },
    },
  });</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Sun Jan 30 2022 18:21:24 GMT+0100 (sentraleuropeisk normaltid)
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    