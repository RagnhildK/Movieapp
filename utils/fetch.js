const baseURL =
  "https://api.themoviedb.org/3/movie/popular?api_key=72c828341c35299683ab545ba90e7f50&language=en-US&page=1";


export const getMovie = async (handleResponse) => {
  try {
    const response = await fetch(baseURL);
    const json = await response.json();
    handleResponse(json);
  } catch (error) {
    console.log(error);
  }
};
