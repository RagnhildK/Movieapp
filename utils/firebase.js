// Import the functions you need from the SDKs you need
import { initializeApp } from "firebase/app";
// import { getAnalytics } from "firebase/analytics";
import {
  getFirestore,
  doc,
  setDoc,
  collection,
  addDoc,
  where,
  query,
  getDoc,
  getDocs,
} from "firebase/firestore";

// TODO: Add SDKs for Firebase products that you want to use
// https://firebase.google.com/docs/web/setup#available-libraries

// Your web app's Firebase configuration
// For Firebase JS SDK v7.20.0 and later, measurementId is optional
const firebaseConfig = {
  apiKey: "AIzaSyCANKrut9VEYA6j9Erq9KEtCgekuNM8Cbs",
  authDomain: "movieapp-903cc.firebaseapp.com",
  projectId: "movieapp-903cc",
  storageBucket: "movieapp-903cc.appspot.com",
  messagingSenderId: "658249651052",
  appId: "1:658249651052:web:b2e8e233e3e95753025949",
  measurementId: "G-C4KJBS2XJY",
};

// Initialize Firebase
const app = initializeApp(firebaseConfig);
// const analytics = getAnalytics(app);

const db = getFirestore();

const UserColl = collection(db, "Users");
// const SessionColl = collection(db, "Participants");

export async function addUser(name) {
  const data = {
    username: name,
  };
  await addDoc(UserColl, data);
}

export async function checkIfUserExists(user) {
  const q = query(UserColl, where("username", "==", user));
  const a = await getDocs(q);
  return a.size != 0;
}

export async function addSession(name, sessionID, nmbMovies) {
  const data = {
    [name]: {},
  };
  const movies = {
    amount: nmbMovies,
  };
  await setDoc(doc(db, "Participants", sessionID), data);
  await setDoc(doc(db, "Movies", sessionID), movies);
}

export async function addParticipant(username, sessionID) {
  const data = {
    [username]: {},
  };
  await setDoc(doc(db, "Participants", sessionID), data, { merge: true });
}

export async function getNmbOfMovies(sessionID, handleResponse) {
  const a = doc(db, "Movies", sessionID);

  let movies;
  await getDoc(a).then((doc) => {
    movies = doc.data();
  });
  await handleResponse(movies);
}

export async function updateRatings(username, ratings, sessionID) {
  const data = {
    [username]: ratings,
  };
  await setDoc(doc(db, "Participants", sessionID), data, { merge: true });
}

export async function getRatings(sessionID, handleResponse) {
  const a = doc(db, "Participants", sessionID);
  let ratings;
  await getDoc(a).then((doc) => {
    ratings = doc.data();
  });
  await handleResponse(ratings);
}

// export async function checkIfUserExists() {
//   const docRef = doc(db, "Users", "");
//   const a = await getDoc();
//   console.log(a);
//   a.forEach((doc) => {
//     // doc.data() is never undefined for query doc snapshots
//     console.log(doc.id, " => ", doc.data());
//   });
// }
