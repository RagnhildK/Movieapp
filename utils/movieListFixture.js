import { configureStore } from "@reduxjs/toolkit";
import movieRatingReducer from "../redux/movieSlicer"

const movies = {
    425909: {
      title: "Ghostbusters: Afterlife",
      posterPath: "/sg4xJaufDiQl7caFEskBtQXfD4x.jpg",
      overview:
        "When a single mom and her two kids arrive in a small town, they begin to discover their connection to the original Ghostbusters and the secret legacy their grandfather left behind.",
    },
    438695: {
      title: "Sing 2",
      posterPath: "/aWeKITRFbbwY8txG5uCj4rMCfSP.jpg",
      overview:
        "Buster and his new cast now have their sights set on debuting a new show at the Crystal Tower Theater in glamorous Redshore City. But with no connections, he and his singers must sneak into the Crystal Entertainment offices, run by the ruthless wolf mogul Jimmy Crystal, where the gang pitches the ridiculous idea of casting the lion rock legend Clay Calloway in their show. Buster must embark on a quest to find the now-isolated Clay and persuade him to return to the stage.",
    },
    460458: {
      title: "Resident Evil: Welcome to Raccoon City",
      posterPath: "/7uRbWOXxpWDMtnsd2PF3clu65jc.jpg",
      overview:
        "Once the booming home of pharmaceutical giant Umbrella Corporation, Raccoon City is now a dying Midwestern town. The company’s exodus left the city a wasteland…with great evil brewing below the surface. When that evil is unleashed, the townspeople are forever…changed…and a small group of survivors must work together to uncover the truth behind Umbrella and make it through the night.",
    },
    512195: {
      title: "Red Notice",
      posterPath: "/lAXONuqg41NwUMuzMiFvicDET9Y.jpg",
      overview:
        "An Interpol-issued Red Notice is a global alert to hunt and capture the world's most wanted. But when a daring heist brings together the FBI's top profiler and two rival criminals, there's no telling what will happen.",
    },
  };

  export const mockStore = configureStore({
    reducer: {
      movieRatings: movieRatingReducer,
    },
    preloadedState: {
      movieRatings: {
        sessionID: "",
        username: "",
        nmbMovies: 0,
        loading: false,
        movies: movies,
        ratings: {},
        totalResults: {},
        sortedIDs: [],
        participants: [],
      },
    },
  });