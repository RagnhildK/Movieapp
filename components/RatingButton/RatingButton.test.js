import React from "react";
import renderer from "react-test-renderer";
import RatingButtons from "./RatingButtons";
import { Provider } from "react-redux";
import { mockStore } from "../../utils/movieListFixture";

it("RatingButtons component snapshot", () => {
  const tree = renderer
    .create(
      <Provider store={mockStore}>
        <RatingButtons id={425909} />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
