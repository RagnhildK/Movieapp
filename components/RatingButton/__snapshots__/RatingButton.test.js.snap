// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingButtons component snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "column",
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 25,
              "margin": 3,
              "tintColor": "#f1c40f",
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
              "width": 25,
            }
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 25,
              "margin": 3,
              "tintColor": "#f1c40f",
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
              "width": 25,
            }
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 25,
              "margin": 3,
              "tintColor": "#f1c40f",
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
              "width": 25,
            }
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 25,
              "margin": 3,
              "tintColor": "#BDC3C7",
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
              "width": 25,
            }
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 25,
              "margin": 3,
              "tintColor": "#BDC3C7",
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
              "width": 25,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
